### ISSUE TITLE
FileNotFoundError: [Errno 2] No such file or directory: '../processed_data/amp_balance enlarge train 0_45_r340.pkl'

### ISSUE THREAD

**Issue by puzhou-wrj** [SJTUBME-QianLab/GBM-GNN#1]
I'm interested in your work and would like to cite it in my paper, and I'm in trouble. I want to know how to put my data in directory to avoid FileNotFoundError: [Errno 2] No such file or directory: '. /processed_data/amp_balance enlarge train 0_45_r340.pkl'

**Comment by Junwaa** [comment]
Thanks for your interest. Owing to patient-privacy considerations, our data are not publicly available. 

You can refer to the code to make your own file.

```python
import pickle


def save_as_dict(
    img_datasets_person, labels_person, save_path="YOUR_SAVE_PATH", save=True
):
    """
    img_datasets_person: A list for storing each 3D data. [(D,H,W),(D,H,W),...,(D,H,W)]
    labels_person: A list for storing the corresponding labels.
    save_path: save_path
    save: bool
    return: A dictionary for storing 3D data based on labels (Keys) 
    """
    dict_data = {"0": [], "1": []}
    for i in range(len(labels_person)):
        if labels_person[i] == 0:
            dict_data["0"].append(img_datasets_person[i])
        if labels_person[i] == 1:
            dict_data["1"].append(img_datasets_person[i])
    num = 0
    for data_p in dict_data["0"]:
        print(data_p.shape)
        num += 1
        print(num)
    num = 0
    for data_p in dict_data["1"]:
        print(data_p.shape)
        num += 1
        print(num)
    if save:
        with open(save_path + "FILE_NAME.pkl", "wb") as f:
            pickle.dump(dict_data, f)
    return dict_data
```

**Comment by puzhou-wrj** [comment]
Thank you very much, you are a great help to me!
I have another question, does my file should follow the format like this:

```
── data
    └── 0
        ├── person_a.nii.gz
        ├── person_b.nii.gz
        └── ...
    └── 1
        ├── person_c.nii.gz
        ├── person_d.nii.gz
        └── ...
```

Thanks very much!

**Comment by Junwaa** [comment]
Yes, you can also design your own dictionary structure as needed.

**Comment by Xiuxiu21** [comment]
> ```python
> import pickle
> 
> 
> def save_as_dict(
>     img_datasets_person, labels_person, save_path="YOUR_SAVE_PATH", save=True
> ):
>     """
>     img_datasets_person: A list for storing each 3D data. [(D,H,W),(D,H,W),...,(D,H,W)]
>     labels_person: A list for storing the corresponding labels.
>     save_path: save_path
>     save: bool
>     return: A dictionary for storing 3D data based on labels (Keys) 
>     """
>     dict_data = {"0": [], "1": []}
>     for i in range(len(labels_person)):
>         if labels_person[i] == 0:
>             dict_data["0"].append(img_datasets_person[i])
>         if labels_person[i] == 1:
>             dict_data["1"].append(img_datasets_person[i])
>     num = 0
>     for data_p in dict_data["0"]:
>         print(data_p.shape)
>         num += 1
>         print(num)
>     num = 0
>     for data_p in dict_data["1"]:
>         print(data_p.shape)
>         num += 1
>         print(num)
>     if save:
>         with open(save_path + "FILE_NAME.pkl", "wb") as f:
>             pickle.dump(dict_data, f)
>     return dict_data
> ```

我也遇到同样情况 ，请问您问题解决了吗
